perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_m365AllowedLicensesPriorities.java

```diff
@@ -0,0 +1,100 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.*;
+import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.FacilityAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.FacilityAttributesModuleImplApi;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * Checks that value of m365AllowedLicensesPriorities is valid:
+ *   -	does not contain a null key
+ *   -	when a change is made, there is no m365LicenseGroup attribute under the same facility
+ *   	containing a license that no longer exists in this map
+ *
+ * @author Michal Berky <michal.berky@cesnet.cz>
+ */
+public class urn_perun_facility_attribute_def_def_m365AllowedLicensesPriorities extends FacilityAttributesModuleAbstract implements FacilityAttributesModuleImplApi {
+
+	@Override
+	public void checkAttributeSyntax(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongAttributeValueException {
+		LinkedHashMap<String, String> licensesPriorities = attribute.valueAsMap();
+		if(licensesPriorities == null) return;
+
+		for (String key : licensesPriorities.keySet()) {
+			if (key == null) {
+				throw new WrongAttributeValueException("There can't be a null key in: " + attribute);
+			}
+		}
+	}
+
+	@Override
+	public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		LinkedHashMap<String, String> licensesMap = attribute.valueAsMap();
+		List<Resource> resources = fetchResourcesForFacility(perunSession, facility);
```

Comment:
```suggestion
		List<Resource> resources = sess.getPerunBl().getFacilitiesManagerBl().getAssignedResources(sess, facility);
```
no need to complicate this, any exceptions thrown are ok as is